### PR TITLE
[Feat/oku 30] Add review on FE

### PR DESCRIPTION
**Pull request by enricodelarosa**
## Change Type
*Put an `x` in all the boxes that apply
- [] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to change)

## Before & after screenshots or video recordings (Before & After)

**I made the styled components reusable so Login, add game, and add review all use them.

image

image

### CODE REVIEW

File: project_app/pages/createreview.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    const [games, setGames] = useState([]); 
+    const selectedGameName = useRef(null);
+    
+    async function fetchGames() {
+        const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/api/games`);
+        const games = await response.json();
+        setGames(games);
+
+        // Name of game on load
+        selectedGameName.current = games[0].name;
```
Comment (idiglove) [inline project_app/pages/createreview.tsx:25]: lets not use ref, should be state

**Review by idiglove — COMMENTED** [review]
some comments

Also, adding review should look like the one in aintboard.org
Not from admin page

File: project_app/pages/addgame.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import "@coreui/coreui/dist/css/coreui.min.css";
+import Swal from "sweetalert2";
+
+import AdminLayout from '../components/AdminLayout';
+
+import {StyledCFormTextarea, StyledCFormInput, StyledCFormLabel, FormLayout, FormTitle, FormDialog, DialogTitle, FormContainer, DialogText, ActionButtonContainer, ActionButton} from '../components/formstyled';
+
+
+
+export default function admin() {
```
[the remark this answers is not in the document]
Comment (enricodelarosa) [reply project_app/pages/addgame.tsx:13]: Yesss, will change this

File: project_app/pages/createreview.tsx (hunk 1 above)
Comment (enricodelarosa) [reply project_app/pages/createreview.tsx:25]: but would using state trigger re-renders if the state changes?

Comment (idiglove) [reply project_app/pages/createreview.tsx:25]: yes its ok.

**Comment by enricodelarosa** [comment]
image

**Comment by enricodelarosa** [comment]
image

File: project_app/components/formstyled.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+font-weight: 400;
+text-align: left;
+`;
+
+const FormContainer = styled.div`
+padding: 2rem;
+`;
+
+const DialogText = styled.h6`
+color: rgb(255, 255, 255);
```
Comment (idiglove) [inline project_app/components/formstyled.tsx:94]: isn't this just #000?

File: project_app/pages/addgame.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (idiglove) [reply project_app/pages/addgame.tsx:13]: still haven't changed

File: project_app/components/formstyled.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+justify-content: left;
+`;
+
+const ActionButton = styled.button`
+  margin-top: 1rem;
+  background-color: rgb(250, 250, 235);
+  color: rgb(0, 0, 0);
+  font-family: Rubik, sans-serif;
+border-radius: 0.625rem;
+font-size: 1.25rem;
```
Comment (idiglove) [inline project_app/components/formstyled.tsx:111]: just do `export const`

File: project_app/pages/createreview.tsx [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+                    
+
+
+                  
+                  
+
+                </RatingSection>
+
+
+
```
Comment (idiglove) [inline project_app/pages/createreview.tsx:244]: good job on the dices, lets just remove the unnecessary whitespaces

File: project_app/components/Dices.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
...26ca2d319ad)"></ellipse><path d="M679.4,767.7c.1,11.8-33.4,21.5-74.7,21.7s-74.8-9.1-74.9-20.9,33.4-21.5,74.7-21.7S679.3,755.9,679.4,767.7Z" transform="translate(0 0)" fill="url(#b2b38476-82ee-48ff-9a2c-743df1db6983)"></path><path d="M309.4,767.7c.1,11.8-33.4,21.5-74.6,21.7s-74.8-9.1-74.9-20.9,33.4-21.5,74.6-21.7S309.4,755.9,309.4,767.7Z" transform="translate(0 0)" fill="url(#a78ef748-a978-4c79-b8e2-ebd431c7a38a)"></path><path d="M356,820.8c-.1,10.3-30.1,18.5-67,18.3s-66.9-8.8-66.8-19.1,30.1-18.5,67-18.3S356.1,810.5,356,820.8Z" transform="translate(0 0)" fill="url(#a4703342-44f3-4f02-9c16-c282894b835f)"></path><path d="M720.4,820.8c-.1,10.3-30.1,18.5-67,18.3s-66.9-8.8-66.8-19.1,30.1-18.5,67-18.3S720.5,810.5,720.4,820.8Z" transform="translate(0 0)" fill="url(#bcff66ee-f454-43ad-8270-9966cb070e2f)"></path><path d="M776.8,864.5c0,7.7-30,13.8-67,13.6s-66.9-6.7-66.8-14.4,30-13.8,67-13.6S776.8,856.7,776.8,864.5Z" transform="translate(0 0)" fill="url(#be841f24-729a-4323-8875-aba3beacf0da)"></path><path d="M411.3,864.5c0,7.7-30,13.8-67,13.6s-66.9-6.7-66.8-14.4,30-13.8,67-13.6S411.3,856.7,411.3,864.5Z" transform="translate(0 0)" fill="url(#a851cb28-6f7a-4893-9c41-2ad05bf80ff9)"></path></svg>
```
Comment (idiglove) [inline project_app/components/Dices.js:86]: extract the svgs to their own react components, then import them here

File: project_app/components/Dices.js [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    const [viewRating, setViewRating] = useState(0);
+
+    useEffect(() => {
+        setViewRating(rating);
+    }, [])
+
+    const ratings = [1, 2, 3, 4, 5, 6];
+
+    const handleDiceHover = (diceValue) => {
+        console.log('mouse entered')
```
Comment (idiglove) [inline project_app/components/Dices.js:17]: remove console log

File: project_app/components/Dices.js [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+import {useEffect, useState} from 'react';
+
+
+
+export function DiceRating({rating, setRating}) {
+
+    const [viewRating, setViewRating] = useState(0);
+
+    useEffect(() => {
+        setViewRating(rating);
```
Comment (idiglove) [inline project_app/components/Dices.js:11]: hmm im worried about the rerenders in this component. you are setting parent state that controls this component, then sets local state based on parent state.

**Review by idiglove — CHANGES_REQUESTED** [review]

**Review by idiglove — APPROVED** [review]

**Pull request merged**